packages/react-router-dev/config/config.ts

```diff
@@ -275,33 +279,55 @@ let deepFreeze = (o: any) => {
   return o;
 };
 
-export function resolvePublicPath(viteUserConfig: Vite.UserConfig) {
-  return viteUserConfig.base ?? "/";
+type Result<T> =
+  | {
+      ok: true;
+      value: T;
+      error?: undefined;
+    }
+  | {
+      ok: false;
+      value?: undefined;
+      error: string;
+    };
+
+function ok<T>(value: T): Result<T> {
+  return { ok: true, value };
 }
 
-let isFirstLoad = true;
-let lastValidRoutes: RouteManifest = {};
+function err<T>(error: string): Result<T> {
+  return { ok: false, error };
+}
 
-export async function resolveReactRouterConfig({
-  rootDirectory,
-  reactRouterUserConfig,
-  routeConfigChanged,
-  viteUserConfig,
-  viteCommand,
-  routesViteNodeContext,
+async function resolveConfig({
+  root,
+  viteNodeContext,
+  reactRouterConfigFile,
 }: {
-  rootDirectory: string;
-  reactRouterUserConfig: ReactRouterConfig;
-  routeConfigChanged: boolean;
-  viteUserConfig: Vite.UserConfig;
-  viteCommand: Vite.ConfigEnv["command"];
-  routesViteNodeContext: ViteNode.Context;
-}) {
-  let vite = importViteEsmSync();
+  root: string;
+  viteNodeContext: ViteNode.Context;
+  reactRouterConfigFile?: string;
+}): Promise<Result<ResolvedReactRouterConfig>> {
+  let reactRouterUserConfig: ReactRouterConfig = {};
+
+  if (reactRouterConfigFile) {
+    try {
+      let configModule = await viteNodeContext.runner.executeFile(
```

Comment:
Review feedback: We should have our own error for when the file is missing rather than letting it fail within Vite when trying to load it.